napari/_vispy/_tests/test_vispy_labels_layer.py

```diff
@@ -68,21 +72,21 @@ def test_labels_fill_slice(make_napari_viewer, array_type):
     layer.n_edit_dimensions = 3
     QCoreApplication.instance().processEvents()
     layer.fill((1, 10, 10), 13, refresh=True)
-    visual = viewer.window._qt_viewer.layer_to_visual[layer]
+    visual = qt_viewer.layer_to_visual[layer]
```

Comment:
Are these changes necessary? (make_napari_viewer -> manually building a qtviewer) They seem to be going backwards. Do we want a QtViewer fixture instead?

Reply:
> Are these changes necessary?

Yes. Our teardown procedure of `Viewer` is fragile and the main place of segfaults in our test set. I have tried to fix this in #6745, but failed. And I cannot estimate how long it may take to fix this in fixture as the problem comes from the architecture of `Viewer`

Reply:
dome